src/jupyter_contrib_nbextensions/nbextensions/varInspector/main.js

```diff
@@ -129,34 +129,31 @@ function html_table(jsonVars) {
     var kernel_config = cfg.kernels_config[kernelLanguage];
     var varList = JSON.parse(String(jsonVars))
 
-    var shape_str;
+    var shape_str = '';
     var has_shape = false;
-    if (varList.some(list_var => "varShape" in list_var )) { //if any of them have a shape
+    if (varList.some(listVar => "varShape" in listVar && listVar.varShape !== '')) { //if any of them have a shape
         shape_str = '<th >Shape</th>';
         has_shape = true;
     }
     var beg_table = '<div class=\"inspector\"><table class=\"table fixed table-condensed table-nonfluid \"><col /> \
  <col  /><col /><thead><tr><th >X</th><th >Name</th><th >Type</th><th >Size</th>' + shape_str + '<th >Value</th></tr></thead><tr><td> \
- </td></tr>'
-    var nb_vars = varList.length;
-    var shape_col_str;
-    for (var i = 0; i < nb_vars; i++) {
+ </td></tr>';
+    varList.forEach(listVar => {
+        var shape_col_str = '';
```

Comment:
Actually this should be 
```
var shape_col_str = '</td><td>';
```
Otherwise varsize and varContent are concatenated in the case there is no shape (try with a single variable eg `x=10`